<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1350</width>
    <height>800</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1350</width>
    <height>800</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>GUI</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout" stretch="1,0">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,1">
      <item>
       <layout class="QGridLayout" name="settingsLayout">
        <item row="3" column="0" colspan="3">
         <widget class="QGroupBox" name="securityPolicy">
          <property name="title">
           <string>Security rules</string>
          </property>
          <widget class="QWidget" name="gridLayoutWidget_3">
           <property name="geometry">
            <rect>
             <x>20</x>
             <y>20</y>
             <width>321</width>
             <height>174</height>
            </rect>
           </property>
           <layout class="QGridLayout" name="gridLayout_3">
            <item row="3" column="0">
             <widget class="QLabel" name="upperLabel">
              <property name="text">
               <string>Min Occurences of upper characters</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="specialLabel">
              <property name="text">
               <string>Min Occurences of special characters</string>
              </property>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="lowerLabel">
              <property name="text">
               <string>Min Occurences of lower characters</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="digitLabel">
              <property name="text">
               <string>Min Occurences of digit characters</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="lengthLabel">
              <property name="text">
               <string>Min Length</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="MinLenSpin">
              <property name="value">
               <number>8</number>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="MinOccSpecialSpin"/>
            </item>
            <item row="2" column="1">
             <widget class="QSpinBox" name="MinOccDigitsSpin">
              <property name="value">
               <number>1</number>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QSpinBox" name="MinOccUpperSpin">
              <property name="value">
               <number>1</number>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <widget class="QSpinBox" name="MinOccLowerSpin">
              <property name="value">
               <number>1</number>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
        <item row="4" column="0" colspan="3">
         <widget class="QPushButton" name="startButton">
          <property name="text">
           <string>Start</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0" colspan="3">
         <widget class="QGroupBox" name="advancedOptions">
          <property name="title">
           <string>Advanced options (only shown in &quot;result.txt&quot;)</string>
          </property>
          <widget class="QWidget" name="gridLayoutWidget_4">
           <property name="geometry">
            <rect>
             <x>20</x>
             <y>20</y>
             <width>301</width>
             <height>107</height>
            </rect>
           </property>
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="1" column="0">
             <widget class="QCheckBox" name="maxAdvancedCheckBox">
              <property name="text">
               <string>Maximum advanced mask size</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QCheckBox" name="topCheckBox">
              <property name="text">
               <string>Top X answers to display</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QCheckBox" name="maxSimpleCheckBox">
              <property name="text">
               <string>Maximum simple mask size</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="MaxSimpleMaxSpin"/>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="MaxAdvMaskSpin"/>
            </item>
            <item row="2" column="1">
             <widget class="QSpinBox" name="TopXAnswerSpin"/>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="fileLine">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>300</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>1000000</width>
            <height>16777215</height>
           </size>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QPushButton" name="Browser">
          <property name="text">
           <string>Browse</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="fileLabel">
          <property name="text">
           <string>File:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0" colspan="3">
         <widget class="QGroupBox" name="globalOptions">
          <property name="title">
           <string>Options</string>
          </property>
          <widget class="QWidget" name="gridLayoutWidget_2">
           <property name="geometry">
            <rect>
             <x>20</x>
             <y>30</y>
             <width>260</width>
             <height>103</height>
            </rect>
           </property>
           <layout class="QGridLayout" name="gridLayout_2">
            <item row="2" column="1">
             <widget class="QLineEdit" name="RegexlineEdit"/>
            </item>
            <item row="0" column="0">
             <widget class="QRadioButton" name="withcountButton">
              <property name="text">
               <string>Withcount</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QCheckBox" name="regexCheckBox">
              <property name="text">
               <string>Regular expression</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QRadioButton" name="classicButton">
              <property name="text">
               <string>Classic</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="threadNumberSpin">
              <property name="minimum">
               <number>1</number>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label">
              <property name="layoutDirection">
               <enum>Qt::LeftToRight</enum>
              </property>
              <property name="text">
               <string>Number of threads</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QFrame" name="resultFrame">
        <property name="minimumSize">
         <size>
          <width>830</width>
          <height>0</height>
         </size>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3" stretch="0,0,1,1,0">
         <item>
          <widget class="QLabel" name="AnalyzedLabel">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="securityLabel">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="lengthWidget" native="true">
           <property name="minimumSize">
            <size>
             <width>800</width>
             <height>0</height>
            </size>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="charsetWidget" native="true"/>
         </item>
         <item>
          <widget class="QLabel" name="resultLabel">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string/>
        </property>
        <property name="pixmap">
         <pixmap>ressources/logoGREYC.png</pixmap>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string/>
        </property>
        <property name="pixmap">
         <pixmap>ressources/logoENSICAEN.png</pixmap>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string/>
        </property>
        <property name="pixmap">
         <pixmap>ressources/logoCNRS.png</pixmap>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string/>
        </property>
        <property name="pixmap">
         <pixmap>ressources/logoUNICAEN.png</pixmap>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1350</width>
     <height>20</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
